### DISCUSSION TITLE
ifort compiler error

### DISCUSSION THREAD

**Discussion by ravikiran19980** [modern-fortran/neural-fortran#101] [Q&A]
Hello. I have written this code separately but didnot clone the entire repository. I just used the files and modules that are needed for my use. I just want to code a simple feedforward neural network for a linear regression problem. I have compiled my codes with gfortran, it runs perfectly fine. However when i try with ifort(version ifort (IFORT) 19.0.4.243 20190416)
The compilation says 

nf_base_layer.f90: catastrophic error: **Internal compiler error: segmentation violation signal raised** Please report this error along with the circumstances in which it occurred in a Software Problem Report.  Note: File and line given may not be explicit cause of this error.
compilation aborted for nf_base_layer.f90 (code 1)
nf_base_layer_submodule.f90(1): error #7013: This module file was not generated by any release of this compiler.   [NF_BASE_LAYER]
submodule(nf_base_layer) nf_base_layer_submodule

So i dont understand why. I have observed maybe some problem with procedure pointers it is not able to handle. Also i see that you have tested and compiled with ifort - 2021.4 version. Could you let me know what could be the reason

Below are my attached codes:  

[vectorized_2.zip](https://github.com/modern-fortran/neural-fortran/files/9908965/vectorized_2.zip)
I compiled with the following command and it works fine:

gfortran nf_activation.f90 nf_activation_submodule.f90 nf_base_layer.f90 nf_base_layer_submodule.f90    nf_input_1d_layer.f90 nf_input_1d_layer_submodule.f90 nf_dense_layer.f90 nf_dense_layer_submodule.f90 nf_layer.f90 nf_layer_submodule.f90 nf_layer_constructs.f90 nf_layer_constructs_submodule.f90 nf_network.f90 nf_network_submodule.f90 nf_loss.f90 nf_loss_submodule.f90 nf_random.f90 nf_random_submodule.f90 nf_optimizers.f90 data_normalizations.f90 data_normalizations_submodule.f90 sine.f90 nf.f90 

but 

ifort nf_activation.f90 nf_activation_submodule.f90 nf_base_layer.f90 nf_base_layer_submodule.f90    nf_input_1d_layer.f90 nf_input_1d_layer_submodule.f90 nf_dense_layer.f90 nf_dense_layer_submodule.f90 nf_layer.f90 nf_layer_submodule.f90 nf_layer_constructs.f90 nf_layer_constructs_submodule.f90 nf_network.f90 nf_network_submodule.f90 nf_loss.f90 nf_loss_submodule.f90 nf_random.f90 nf_random_submodule.f90 nf_optimizers.f90 data_normalizations.f90 data_normalizations_submodule.f90 sine.f90 nf.f90 

doesnt work. 

gfortran version : GNU Fortran (SUSE Linux) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.

ifort version:ifort (IFORT) 19.0.4.243 20190416
Copyright (C) 1985-2019 Intel Corporation.  All rights reserved.

**Comment by milancurcic** [comment]
Hi @ravikiran19980,

If I remove the .mod and .smod files, the gfortran doesn't build for me either. Can you include some build system instruction for the project (Makefile, CMakeLists.txt, fpm.toml, or similar)?

An internal compiler error is always a compiler bug. You could try reporting it to Intel, but likely the response may be to upgrade to a more recent version.

I'm happy to try building this on my end if you include a build system config that properly orders the compilation of the source files.
